shared/studio/state/database.ts

```diff
@@ -182,6 +185,39 @@ export class DatabaseState extends Model({
     };
   }
 
+  watchForSchemaChanges() {
+    let fetchingDbInfo: Promise<void> | null = null;
+    const listener = async () => {
+      if (window.document.visibilityState === "visible") {
+        const instanceState = instanceCtx.get(this);
+        if (!instanceState || !this.schemaId || fetchingDbInfo !== null)
+          return;
+
+        fetchingDbInfo = instanceState.fetchDatabaseInfo();
+        await fetchingDbInfo;
+        fetchingDbInfo = null;
+
+        const dbInfo = instanceState.databases?.find(
+          (db) => db.name === this.name
+        );
+
+        if (dbInfo && dbInfo.last_migration != this.schemaLastMigration) {
+          this.fetchSchemaData();
```

Comment:
Is this sufficient to also trigger a refetch of the _data_ as well?

Reply:
Yes, the schema data is obserable, and the data explorer already watches for changes and will trigger a refresh of the columns and data if it changes.